sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -501,32 +501,44 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
   ): Promise<UpdateResult> {
     const auditUser = { id: auditUserId } as User;
     const now = new Date();
-    const updateResult = await this.repo
-      .createQueryBuilder()
-      .update(DisbursementSchedule)
-      .set({
-        coeStatus: COEStatus.declined,
-        coeUpdatedBy: auditUser,
-        coeUpdatedAt: now,
-        coeDeniedReason: { id: coeDeniedReasonId },
-        coeDeniedOtherDesc:
-          coeDeniedReasonId === COE_DENIED_REASON_OTHER_ID
-            ? otherReasonDesc
-            : null,
-        modifier: auditUser,
-        updatedAt: now,
-      })
-      .where("id = :disbursementScheduleId", { disbursementScheduleId })
-      .andWhere("coeStatus = :required", { required: COEStatus.required })
-      .execute();
+    return await this.dataSource.transaction(
+      async (transactionalEntityManager) => {
+        const updateResult = await transactionalEntityManager
+          .getRepository(DisbursementSchedule)
+          .createQueryBuilder()
+          .update(DisbursementSchedule)
+          .set({
+            coeStatus: COEStatus.declined,
+            coeUpdatedBy: auditUser,
+            coeUpdatedAt: now,
+            coeDeniedReason: { id: coeDeniedReasonId },
+            coeDeniedOtherDesc:
+              coeDeniedReasonId === COE_DENIED_REASON_OTHER_ID
+                ? otherReasonDesc
+                : null,
+            modifier: auditUser,
+            updatedAt: now,
+          })
+          .where("id = :disbursementScheduleId", { disbursementScheduleId })
+          .andWhere("coeStatus = :required", { required: COEStatus.required })
+          .execute();
 
-    if (updateResult.affected !== 1) {
-      throw new Error(
-        `While updating COE status to '${COEStatus.declined}' the number of affected row was bigger than the expected one. Expected 1 received ${updateResult.affected}`,
-      );
-    }
+        if (updateResult.affected !== 1) {
+          throw new Error(
+            `While updating COE status to '${COEStatus.declined}' the number of affected row was bigger than the expected one. Expected 1 received ${updateResult.affected}`,
```

Comment:
I would change from `row was bigger` to something that just states that it is different, in this way, if zero records were affected the message would still fit.